### PR TITLE
Add a /tasks API

### PR DESCRIPTION
**Pull request by agrare**
Add an API for Tasks and update swagger.yaml

```
http://localhost:3000/api/v0.0/tasks
[{"id":"1","name":"Order Service Plan","status":null,"state":null,"message":null,"progress":null,"tenant_id":1,"created_at":"2018-10-26T20:08:34.643Z","updated_at":"2018-10-26T20:08:34.643Z"}]

http://localhost:3000/api/v0.0/tasks/1
{"id":"1","name":"Order Service Plan","status":null,"state":null,"message":null,"progress":null,"tenant_id":1,"created_at":"2018-10-26T20:08:34.643Z","updated_at":"2018-10-26T20:08:34.643Z"}
```

~~Depends: https://github.com/ManageIQ/topological_inventory-core/pull/36~~

### CODE REVIEW

File: app/controllers/api/v0/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module Api
+  module V0
+    class TasksController < ApplicationController
+      include Api::Mixins::IndexMixin
+      include Api::Mixins::ShowMixin
+
+      def list_params
+        params.permit(:tenant_id)
```
Comment (Fryguy) [inline app/controllers/api/v0/tasks_controller.rb]: permit or require?

File: public/doc/swagger-2-v0.0.1.yaml [hunk 2]
```diff
@@ -781,6 +781,41 @@ paths:
...
+      description: Returns a Task object
+      produces:
+      - application/json
+      parameters:
+      - name: id
+        in: path
+        description: ID of the resource to return
+        required: true
+        type: string
+        format: uuid
```
Comment (Fryguy) [inline public/doc/swagger-2-v0.0.1.yaml]: id is a uuid?

File: app/controllers/api/v0/tasks_controller.rb (hunk 1 above)
Comment (bdunne) [reply app/controllers/api/v0/tasks_controller.rb]: I think this needs to be figured out generically.  It's supposed to be in a header, right?

File: public/doc/swagger-2-v0.0.1.yaml (hunk 2 above)
Comment (bdunne) [reply public/doc/swagger-2-v0.0.1.yaml]: not actually

Comment (agrare) [reply public/doc/swagger-2-v0.0.1.yaml]: Yeah I thought uuid was odd but every other `id` was `format: uuid` so I figured it was some swagger thing.  E.g.: https://github.com/ManageIQ/topological_inventory-api/blob/f29276eb7a40ed85642ef668cc71a17cc2236a4f/public/doc/swagger-2-v0.0.1.yaml#L38-L44

**Comment by agrare** [comment]
@bdunne PTAL

**Pull request merged**